src/main/java/com/kodilla/ecommercee/controller/CartController.java

```diff
@@ -1,34 +1,68 @@
+import com.kodilla.ecommercee.mapper.CartMapper;
+import com.kodilla.ecommercee.mapper.ProductMapper;
+import com.kodilla.ecommercee.service.CartService;
+import com.kodilla.ecommercee.service.ProductService;
+import org.springframework.beans.factory.annotation.Autowired;
 import org.springframework.http.MediaType;
+
 import org.springframework.web.bind.annotation.*;
 
 import java.util.ArrayList;
 import java.util.List;
 
+@CrossOrigin(origins = "*")
 @RestController
-@RequestMapping("api/v1/ecommercee/carts")
+@RequestMapping("api/v1/ecommercee")
 public class CartController {
+    @Autowired
+    CartService cartService;
+
+    @Autowired
+    ProductService productService;
+
+    @Autowired
+    CartMapper cartMapper;
+
+    @Autowired
+    ProductMapper productMapper;
+
     @PostMapping(consumes = MediaType.APPLICATION_JSON_VALUE, produces = MediaType.APPLICATION_JSON_VALUE)
     public void newCart(@RequestBody CartDto cartDto) {
+        cartService.saveCart(cartMapper.mapToCart(cartDto));
     }
 
     @GetMapping(path = "/{id}", consumes = MediaType.APPLICATION_JSON_VALUE)
-    public List<ProductDto> getProducts(@PathVariable Long cartId) {
-        return new ArrayList<ProductDto>();
+    public List<ProductDto> getProducts(@PathVariable("id") Long cartId)
+            throws NotFoundException {
+        List<ProductDto> productDtos = new ArrayList<>();
+        cartService.getCartById(cartId).orElseThrow(NotFoundException::new)
+                .getProducts()
+                .forEach(product -> productDtos.add(productMapper.mapToProductDto(product)));
+        return productDtos;
     }
 
-    @PutMapping(consumes = MediaType.APPLICATION_JSON_VALUE)
-    public void updateCart(@PathVariable Long cartId, @RequestBody ProductDto productDto) {
+    @PutMapping(path = "/{id}", consumes = MediaType.APPLICATION_JSON_VALUE)
+    public void addProduct(@PathVariable("id") Long cartId, @RequestBody ProductDto productDto)
+            throws NotFoundException {
+        cartService.getCartById(cartId).orElseThrow(NotFoundException::new)
+                .getProducts()
+                .add(productMapper.mapToProduct(productDto));
     }
 
-    @DeleteMapping(path = "/{id}")
-    public void deleteProduct(@PathVariable Long cartId, @PathVariable Long productId) {
+    @DeleteMapping(path = "/{cart_id}&{product_id}")
+    public void deleteProduct(@PathVariable("cart_id") Long cartId, @PathVariable("product_id") Long productId)
+            throws NotFoundException {
+        cartService.getCartById(cartId).orElseThrow(NotFoundException::new)
+                .getProducts()
+                .remove(productService.getProduct(productId));
     }
 
     @PostMapping(path = "/{id}", consumes = MediaType.APPLICATION_JSON_VALUE, produces = MediaType.APPLICATION_JSON_VALUE)
-    public void createOrder(@PathVariable Long cartId) {
+    public void createOrder(@PathVariable("id") Long cartId) {
+
```

Comment:
Brakuje jeszcze implementacji dla createOrder()

Reply:
Wg mnie w tej chwili trudno coś sensownego wstawić, brakuje pełnego OrderControllera (jeśli poprawnie to rozumiem).

Reply:
Możesz napisać, IntelliJ podświetli na czerwono najwyżej, że nie ma takich klas lub metod, ale sprawdzimy ogólną logikę